### PR TITLE
[origin] Add options to control oc adm diagnostics command invocation

### PR DESCRIPTION
**Pull request by codificat**
Add options to control the invocation of the "oc adm diagnostics" command.

These are useful in certain deployments that can't or don't want to have sosreport invoke the command (the `diag` option), or at least need to prevent the creation of test pods (the `diag-prevent` option).

Defaults are set to maintain the current behaviour.

Also taking the opportunity to remove the only reference to `oadm` in the node config check (replaced by `oc adm`) and clean up `oc` invocation on nodes to remove reference to the admin kubeconfig (which is only present on masters).

Fixes [RHBZ#1463509](https://bugzilla.redhat.com/show_bug.cgi?id=1463509).

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?

### CODE REVIEW

**Comment by myllynen** [comment]
As said in the RHBZ, this sounds very good, thanks a lot for working on this!

**Comment by pmoravec** [comment]
+1

File: sos/plugins/origin.py [hunk 1]
```diff
@@ -14,8 +14,8 @@
 # with this program; if not, write to the Free Software Foundation, Inc.,
 # 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
 
-from sos.plugins import Plugin, RedHatPlugin
 import os.path
+from sos.plugins import Plugin, RedHatPlugin
```
Comment (bmr-cymru) [inline sos/plugins/origin.py]: Style note: typically we import the sos bits first, then python libs - I'll fix this up on merge.

File: sos/plugins/origin.py [hunk 2]
```diff
@@ -45,14 +45,21 @@ class OpenShiftOrigin(Plugin):
     files = None  # file lists assigned after path setup below
     profiles = ('openshift',)
 
+    option_list = [
+        ("diag", "run 'oc adm diagnostics' to collect its output",
+         'fast', True),
+        ("diag-prevent", "set --prevent-modifications on 'oc adm diagnostics'",
+         'fast', False),
```
Comment (bmr-cymru) [inline sos/plugins/origin.py]: Wouldn't we actually want this `True` by default? Are there any harmful modifications possible is this option is absent?

Comment (codificat) [reply sos/plugins/origin.py]: There was a discussion about this in [bug 1463509](https://bugzilla.redhat.com/show_bug.cgi?id=1463509#c13) and the conclusion in comment 13 was that we'd introduce these options but leave their defaults to match the current behaviour.

The modifications that the option refers to involve the creation of temporary namespaces and pods to help diagnose network connectivity, and they are expected to be "safe enough" and hence are enabled by default (i.e. `--prevent-modifications` is off by default).

Basically the new option is here to allow disabling when/if these diagnostics can cause a problem (e.g. if the cluster is running disconnected and doesn't have access to the images required to run these diagnostics).

Comment (bmr-cymru) [reply sos/plugins/origin.py]: Thanks for confirming, I'll defer to that judgement in that case.

**Comment by bmr-cymru** [comment]
I've dropped the import shuffling from the patch as I merged - unless there is some specific reason it's normally preferred to have the `sos` imports first, and then other auxiliary python modules.
